*** /home/ash/GSoC/test-install/nova/nova/compute/api.py	2014-08-12 17:03:41.625096932 +0400
--- /home/ash/GSoC/nova/nova/compute/api.py	2014-08-14 16:56:41.890782813 +0400
***************
*** 693,699 ****
                                           requested_networks, config_drive,
                                           block_device_mapping,
                                           auto_disk_config, reservation_id,
!                                          max_count):
          """Verify all the input parameters regardless of the provisioning
          strategy being performed.
          """
--- 693,699 ----
                                           requested_networks, config_drive,
                                           block_device_mapping,
                                           auto_disk_config, reservation_id,
!                                          max_count,topology_priority):
          """Verify all the input parameters regardless of the provisioning
          strategy being performed.
          """
***************
*** 777,783 ****
              'availability_zone': availability_zone,
              'root_device_name': root_device_name,
              'progress': 0,
!             'system_metadata': system_metadata}
  
          options_from_image = self._inherit_properties_from_image(
                  boot_meta, auto_disk_config)
--- 777,784 ----
              'availability_zone': availability_zone,
              'root_device_name': root_device_name,
              'progress': 0,
!             'system_metadata': system_metadata,
!             'topology_priority': topology_priority}
  
          options_from_image = self._inherit_properties_from_image(
                  boot_meta, auto_disk_config)
***************
*** 924,930 ****
                 requested_networks, config_drive,
                 block_device_mapping, auto_disk_config,
                 reservation_id=None, scheduler_hints=None,
!                legacy_bdm=True, shutdown_terminate=False):
          """Verify all the input parameters regardless of the provisioning
          strategy being performed and schedule the instance(s) for
          creation.
--- 925,932 ----
                 requested_networks, config_drive,
                 block_device_mapping, auto_disk_config,
                 reservation_id=None, scheduler_hints=None,
!                legacy_bdm=True, shutdown_terminate=False,
!                topology_priority=None):
          """Verify all the input parameters regardless of the provisioning
          strategy being performed and schedule the instance(s) for
          creation.
***************
*** 964,970 ****
                  forced_host, user_data, metadata, injected_files, access_ip_v4,
                  access_ip_v6, requested_networks, config_drive,
                  block_device_mapping, auto_disk_config, reservation_id,
!                 max_count)
  
          # max_net_count is the maximum number of instances requested by the
          # user adjusted for any network quota constraints, including
--- 966,972 ----
                  forced_host, user_data, metadata, injected_files, access_ip_v4,
                  access_ip_v6, requested_networks, config_drive,
                  block_device_mapping, auto_disk_config, reservation_id,
!                 max_count,topology_priority)
  
          # max_net_count is the maximum number of instances requested by the
          # user adjusted for any network quota constraints, including
***************
*** 988,994 ****
  
          filter_properties = self._build_filter_properties(context,
                  scheduler_hints, forced_host, forced_node, instance_type)
! 
          self._update_instance_group(context, instances, scheduler_hints)
  
          for instance in instances:
--- 990,997 ----
  
          filter_properties = self._build_filter_properties(context,
                  scheduler_hints, forced_host, forced_node, instance_type)
!         if topology_priority is not None:
!             filter_properties['instance_type']['topology_priority'] = topology_priority
          self._update_instance_group(context, instances, scheduler_hints)
  
          for instance in instances:
***************
*** 1310,1316 ****
                 block_device_mapping=None, access_ip_v4=None,
                 access_ip_v6=None, requested_networks=None, config_drive=None,
                 auto_disk_config=None, scheduler_hints=None, legacy_bdm=True,
!                shutdown_terminate=False):
          """Provision instances, sending instance information to the
          scheduler.  The scheduler will determine where the instance(s)
          go and will handle creating the DB entries.
--- 1313,1319 ----
                 block_device_mapping=None, access_ip_v4=None,
                 access_ip_v6=None, requested_networks=None, config_drive=None,
                 auto_disk_config=None, scheduler_hints=None, legacy_bdm=True,
!                shutdown_terminate=False, topology_priority=None):
          """Provision instances, sending instance information to the
          scheduler.  The scheduler will determine where the instance(s)
          go and will handle creating the DB entries.
***************
*** 1340,1346 ****
                                 block_device_mapping, auto_disk_config,
                                 scheduler_hints=scheduler_hints,
                                 legacy_bdm=legacy_bdm,
!                                shutdown_terminate=shutdown_terminate)
  
      def trigger_provider_fw_rules_refresh(self, context):
          """Called when a rule is added/removed from a provider firewall."""
--- 1343,1350 ----
                                 block_device_mapping, auto_disk_config,
                                 scheduler_hints=scheduler_hints,
                                 legacy_bdm=legacy_bdm,
!                                shutdown_terminate=shutdown_terminate,
!                                topology_priority=topology_priority)
  
      def trigger_provider_fw_rules_refresh(self, context):
          """Called when a rule is added/removed from a provider firewall."""
