*** /home/ash/GSoC/test-install/nova/nova/db/api.py	2014-08-12 17:03:41.789087451 +0400
--- /home/ash/GSoC/nova/nova/db/api.py	2014-08-12 17:21:01.616971557 +0400
***************
*** 127,132 ****
--- 127,177 ----
  
  ###################
  
+ """
+ TopologyWeigher references to the db calls
+ start
+ """
+ 
+ def traffic_get(context):
+     return IMPL.traffic_get(context)
+ 
+ def traffic_get_avg(context,time):
+     return IMPL.traffic_get_avg(context,time)
+ 
+ def traffic_add(context,values):
+     return IMPL.traffic_add(context,values)
+ 
+ def ping_get(context):
+     return IMPL.ping_get(context)
+ 
+ def ping_get_avg(context,time):
+     return IMPL.ping_get_avg(context,time)
+ 
+ def ping_add(context,values):
+     return IMPL.ping_add(context,values)
+ 
+ def node_add(context,values):
+     return IMPL.node_add(context,values)
+ 
+ def edge_add(context,values):
+     return IMPL.edge_add(context,values)
+ 
+ def node_get(context):
+     return IMPL.node_get(context)
+ 
+ def edge_get(context):
+     return IMPL.edge_get(context)
+ 
+ def check_node(context):
+     return IMPL.check_node(context)
+ 
+ def check_edge(context):
+     return IMPL.check_edge(context)
+ 
+ """
+ end
+ """
+ 
  
  def service_destroy(context, service_id):
      """Destroy the service or raise if it does not exist."""
