*** /home/ash/GSoC/test-install/nova/nova/scheduler/filter_scheduler.py	2014-08-12 17:03:44.884908452 +0400
--- /home/ash/GSoC/nova/nova/scheduler/filter_scheduler.py	2014-08-14 18:00:01.622999625 +0400
***************
*** 236,241 ****
--- 236,243 ----
          """Returns a list of hosts that meet the required specs,
          ordered by their fitness.
          """
+ 
+         topology_priority = filter_properties['instance_type'].get('topology_priority',None)
          elevated = context.elevated()
          instance_properties = request_spec['instance_properties']
          instance_type = request_spec.get("instance_type", None)
***************
*** 253,258 ****
--- 255,262 ----
  
          self.populate_filter_properties(request_spec,
                                          filter_properties)
+         if topology_priority is not None:
+             filter_properties['instance_type']['topology_priority'] = topology_priority
  
          # Find our local list of acceptable hosts by repeatedly
          # filtering and weighing our options. Each time we choose a
